containers/src/Kokkos_Vector.hpp

```diff
@@ -147,6 +147,72 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   void clear() { _size = 0; }
 
+  void insert(const_iterator it, const value_type& val) { insert(it, 1, val); }
+
+  void insert(const_iterator it, size_type count, const value_type& val) {
+    DV::sync_host();
+    DV::modify_host();
+    if (it < begin() || it >= end())
```

Comment:
Too restrictive.  Wouldn't let you insert in empty vector or at the back of a vector.
```suggestion
    if (it < begin() || it > end())
```

Reply:
Yeah looks like you are allowed to insert at `end()` so this needs fixing too.